webtop/__init__.py

```diff
@@ -50,41 +54,42 @@ def parse_args() -> argparse.Namespace:
 
 
 def are_args_valid(args: argparse.Namespace) -> bool:
-    return all(
-        (args.url.startswith(("http://", "https://")), args.request_history >= 1, args.timeout > 0, args.threads > 0)
-    )
+    return all((args.url.is_absolute(), args.request_history >= 1, args.timeout > 0, args.jobs > 0))
 
 
 class Result(object):
-    def __init__(self, *, response: Optional[requests.models.Response] = None, error: Optional[Exception] = None):
+    def __init__(self, *, response: Optional[aiohttp.ClientResponse] = None, error: Optional[Exception] = None):
         self.response = response
         self.error = error
-
         if self.response is None or self.error is not None:
             self.is_success = False
         else:
-            self.is_success = self.response.status_code >= 200 and self.response.status_code < 400
+            self.is_success = 200 <= self.response.status < 400
 
 
 class ResponseResult(Result):
-    def __init__(self, *, response: requests.models.Response):
+    def __init__(self, *, response: aiohttp.ClientResponse, start_time: float, end_time: float):
```

Comment:
Would it be simpler to have a `duration` arg here and do the math prior to constructing the result?

Reply:
I considered it and I can't remember my reason for doing it this way, so I'll refactor this.